packages/components/src/styles/components/flyout.scss

```diff
@@ -17,18 +17,20 @@
 }
 
 .hds-flyout {
-  z-index: 49; // modal component is 50; we want to make sure the flyout always stays below the modal, regardless the DOM order
+  position: fixed;
+  z-index: 49; // modal component is 50; we want to make sure the flyout always stays below the modal, regardless of DOM order
   flex-direction: column;
+  box-sizing: inherit;
   height: 100vh;
   max-height: 100vh;
   margin: 0;
   padding: 0;
   background: var(--token-color-surface-primary);
   border: none;
   box-shadow: 0 2px 3px 0 #3b3d4540, 0 12px 24px 0 #3b3d4559; // TODO: convert to token
+  inset: 0;
```

Comment:
nitpick: I would move it under the `z-index` declarations so it's near the "position/layout" block (they're related)